text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+Alternatively, the custom element name can be stored in a `weakMap` and a public API can be exposed to retrieve the constructor's custom element name. However, because the custom element name can be used as a selector with `this.template.querySelector` placing it on the constructor seems more ergonomic.
+
+#### Defining an anchor in the DOM
+
+##### `<lwc.dynamic>` tag
+
+`<lwc.dynamic>` is a placeholder tag for the dynamic component and will not be rendered to the DOM.  It serves as a special signal to the compiler that the dynamic component will be rendered at the specific location in the DOM. (Similar to how `template` is not rendered)
+
+Note the separation of the namespace and name using a '.' is intentional because of the following reasons:
+
+1. It is neither a valid [tag name](https://html.spec.whatwg.org/multipage/syntax.html#syntax-tag-name) nor a valid [custom element name](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) which should clearly signal that this is a special element (which will not be rendered to the DOM).
+2. Initially considered using a ':' instead of a '.' but ':' is the delimiter Aura uses to separate namespace and name.  To prevent confusion on whether this is an Aura tag, decided to use '.' instead.
+
+Additionally, `<lwc.dynamic>` is intended to be used in tandem with `lwc:is`, otherwise `<lwc:dynamic>` serves no purpose. A warning should be issued when `<lwc.dynamic>` is used without the `lwc:is` template directive.
+
+##### `lwc:is` directive
+
+A new template directive called `lwc:is` will be used to signal to compiler that the constructor is unknown at compile time. 
+
+The directive has the following properties:
+
+1. It can only be used with `<lwc.dynamic>` tag.
+    - An error will be issued when `lwc:is` is used on any other element.
+2. At compile time the value supplied to `lwc:is` can only be an expression.
+3. At runtime the expression supplied to `lwc:is` must be a `LightningElementConstructor`.
+    - An error will be thrown if anything other than a `LightningElementConstructor` is passed to `lwc:is`.
+
+#### Instantiation of the custom element
+
+Dynamic component instantiation follows the same creation path as any other LWC custom element. When the dynamic component is ready to be rendered, the element will be created and mounted to the DOM.  If there is an existing element supplied to `lwc:is`, it will be unmounted as well.
+
+This means that when the value provided to `lwc:is` changes the component and all of its children will be destroyed and recreated using the new constructor.  In addition, when the value provided to `lwc:is` is evaluated, the LWC engine will render depending on the following scenarios:
+
+- **The constructor is `undefined` or `null`**
+    - When the constructor is undefined the `<lwc.dynamic>` tag along with all of its children are not rendered to the DOM.
```

Comment:
I would align with Svelte and say "when it is falsy." This handles cases like `valueThatMayBeFalse && MyConstructor`.